paasta_tools/marathon_tools.py

```diff
@@ -759,10 +760,16 @@ def get_classic_services_that_run_here():
             os.path.exists(os.path.join(PUPPET_SERVICE_DIR, i))
         }
 
+    puppet_service_dir_namespaces = {}
+    for service in puppet_service_dir_services:
```

Comment:
I think the way this is organized is a little unintuitive.

I think it would be easier to understand if there were 3 functions that were concatenated:
* `get_classic_services_that_run_here`
* `get_puppet_services_that_run_here`
* `get_marathon_services_that_run_here`

I think if it was split up like this, then there wouldn't be to functions that sound really similar (get_classic_services_that_run_here, get_classic_services_running_here_for_nerve)
I think the confusion comes from the fast that you need both of these functions to correlate the puppet stuff together. I bet if the puppet parts were in their own thing, then you wouldn't have to do the whole "tuple" return thing.

Can you think about if this is better? If it does, please try it out, if not, I'm totally fine with working code.